<launch>

<arg name="width" default="800" />
<arg name="height" default="600" />
<arg name="fps" default="15" />
<arg name="format" default="YUYV"/>

<arg name="frame_left" default="stereo_left_frame" />
<arg name="device_left" default="/dev/video0" />
<arg name="camera_info_url_left" default="file://$(find dvrk_vision)/defaults/stereo_cam0_calibration.yaml"/>
<arg name="output_topic_left" default="image_raw" />

<node pkg="rviz" type="rviz" name="frame_visualizer"
      args="-d $(find dvrk_vision)/defaults/default.rviz" />

<!-- Static Camera transform publisher -->
<node pkg="tf" type="static_transform_publisher" name="cam_tf_publisher_l" args="0.0 0.0 0.0 0.0 3.14 0.0  world stereo_left_frame 100"/>

<!-- Static Camera transform publisher -->
<node pkg="tf" type="static_transform_publisher" name="cam_tf_publisher_r" args="-0.06172 0.0 0.0 0.0 3.14 0.0  world stereo_right_frame 100"/>

<node ns="stereo/left" pkg="uvc_camera" type="uvc_camera_node" name="uvc_camera_stereo_left" output="screen">
<param name="width" value="$(arg width)" />
<param name="height" value="$(arg height)" />
<param name="fps" value="$(arg fps)" />
<param name="frame_id" value="$(arg frame_left)" />
<param name="format" value="$(arg format)"/>
<param name="device" value="$(arg device_left)" />    
<param name="camera_info_url" value="$(arg camera_info_url_left)" />
<param name="output_topic" value="$(arg output_topic_left)" />
<param name="white_balance_temperature_auto" value="false"/>
<param name="exposure_auto" value="0"/>
</node>

<arg name="marker_size" default="4.4" />
<arg name="max_new_marker_error" default="0.08" />
<arg name="max_track_error" default="0.2" />
<arg name="cam_info_topic" default="camera_info"/>
<arg name="output_frame" default="/stereo_left_frame" />

<node ns="stereo/left" name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
	<param name="marker_size"           type="double" value="$(arg marker_size)" />
	<param name="max_new_marker_error"  type="double" value="$(arg max_new_marker_error)" />
	<param name="max_track_error"       type="double" value="$(arg max_track_error)" />
	<param name="output_frame"          type="string" value="$(arg output_frame)" />

	<remap from="camera_image"  to="$(arg output_topic_left)" />
	<remap from="camera_info"   to="$(arg cam_info_topic)" />
</node>

<node ns="stereo/left/" pkg="image_proc" type="image_proc" name="image_processing" output="screen"/>

</launch>